### PR TITLE
Fixed scroll problems, maxFixtures prop, new ref pattern

### PR DESCRIPTION
**Pull request by mskec**
### Description

- added `maxFixtures` prop to allow configuration of the number of fixtures being shown. Default is set to `10` which was hardcoded so far.
- Updating refs to new pattern
- When there is a lot of fixtures and scroll appears, if you use arrow keys to move through the list you can move to the item which is not visible due to scroll. Fixed it by scrolling item to center if item is not visible.
- small modification of `.eslintrc`

### Checklist

- [x] All tests passing
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary
- [x] Commits and PR follow conventions

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -235,6 +234,8 @@
     "prefer-reflect": 1,
     "prefer-spread": 1,
     "require-yield": 2,
-    "react/jsx-no-bind": 2
+    "react/jsx-no-bind": [2, {
+      "ignoreRefs": true
+    }]
   }
```
Comment (ro-ka) [inline .eslintrc]: Please do not make ESLint changes. We have a company wide ruleset on all our projects.

File: src/Geosuggest.jsx [hunk 2]
```diff
@@ -384,7 +384,7 @@ class Geosuggest extends React.Component {
       ),
       shouldRenderLabel = this.props.label && attributes.id,
       input = <Input className={this.props.inputClassName}
-        ref='input'
+        ref={i => this.input = i}
```
Comment (ro-ka) [inline src/Geosuggest.jsx]: I like this improvement. Can you create a independent PR with this?

File: README.md [hunk 3]
```diff
@@ -136,6 +136,12 @@ An array with fixtures (defaults). Each fixture has to be an object with a `labe
 
 You can also add a `className` key to a fixture. This class will be applied to the fixture item.
 
+### maxFixtures
+Type: `Number`
+Default: `10`
+
+Maximum number of fixtures to render.
```
Comment (ro-ka) [inline README.md]: Why make this configurable? Any specific use case? Google for example only returns five results.

File: src/suggest-item.jsx [hunk 4]
```diff
@@ -19,6 +19,32 @@ export default class SuggestItem extends React.Component {
...
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isActive && !this.props.isActive) {
+      this.scrollIfNeeded();
+    }
+  }
+
+  /**
+   * Scrolling current item to the center of the list if item needs scrolling.
+   * Item is scrolled to the center of the list.
+   */
```
Comment (ro-ka) [inline src/suggest-item.jsx]: Can you explain what this fixes / adds? Why is this added?

File: README.md (hunk 3 above)
Comment (mskec) [inline README.md]: Yes google returns only 5, but you can have predefined default fixture set of more than 10.
Without this configuration you are limited to 10 predefined fixtures.

File: src/suggest-item.jsx (hunk 4 above)
Comment (mskec) [inline src/suggest-item.jsx]: I wrote it in description
c/p
_When there is a lot of fixtures and scroll appears, if you use arrow keys to move through the list you can move to the item which is not visible due to scroll. Fixed it by scrolling item to center if item is not visible._

Comment (ro-ka) [inline src/suggest-item.jsx]: Oh, sorry, missed that.

Comment (mskec) [inline src/suggest-item.jsx]: You can find demo here: https://mskec.github.io/react-geosuggest/
1. click input
2. use arrow keys to change fixtures, list won't scroll when you go to fixture 6.
3. use mouse scroll to see that there are more fixtures

File: README.md (hunk 3 above)
Comment (mskec) [inline README.md]: In the demo I provided below, I have defined 13 fixtures, but only 10 are rendered. This configuration solves that problem.

File: .eslintrc (hunk 1 above)
Comment (ro-ka) [inline .eslintrc]: Change just that one rule then, not others.

File: README.md (hunk 3 above)
Comment (ro-ka) [inline README.md]: It means 10 are rendered when nothing is entered. You can easily narrow down the list by typing something in the input.

File: src/suggest-item.jsx (hunk 4 above)
Comment (ro-ka) [inline src/suggest-item.jsx]: Thanks! Indeed didn’t notice.

Comment (ro-ka) [inline src/suggest-item.jsx]: Limit it to the line length of 80, then it’s fine.
